### PR TITLE
 #17 fix curl_reset for php <5.5

### PR DESCRIPTION
**Pull request by alexbarylo**
Sorry, your latest commit didn't help -- http://screencast.com/t/ONsuinEnoREZ

### CODE REVIEW

File: Services/Curl.php [hunk 1]
```diff
@@ -95,7 +95,9 @@ public function sendRequest($url, $method, array $options = array(), $payload =
         $curlMetaData = (object) curl_getinfo($this->curl);
 
         $this->curlOptionsHandler->reset();
-        function_exists(curl_reset($this->curl)) ? curl_reset($this->curl) : curl_setopt_array($this->curl, $this->curlOptionsHandler->getOptions());
+        if (function_exists('curl_reset')) {
```
Comment (passi246) [inline Services/Curl.php]: There is no reset functionality for the case the "curl_reset" does not exist. Use the ternary operator like suggested, excepting the wrong usage of function_exists.

Comment (alexbarylo) [inline Services/Curl.php]: sorry, but for me this code
curl_setopt_array($this->curl, $this->curlOptionsHandler->getOptions());
is not the same as curl_reset, or am I wrong?

Comment (passi246) [inline Services/Curl.php]: curl_reset is used for resetting the curl to the default state, so curl_setopt_array() is used for resetting the custom curl options of the current curl object (kind of a soft reset). In my opinion there is no need to reset other stuff than the custom options to the default ones, if the curl_reset function does not exist.. If you see another part that needs to be reseted let us know...

Comment (alexbarylo) [inline Services/Curl.php]: sorry, you are right.
I lost previous line 
>>$this->curlOptionsHandler->reset();
here is the end of the day in Ukraine =)
please check it.

**Comment by TobiasHauck** [comment]
This one is correct now, but I already fixed it. Should work now. The only reason it didn't work before is that I didn't use the function_exists correctly. Sorry :D

File: Services/Curl.php (hunk 1 above)
Comment (TobiasHauck) [inline Services/Curl.php]: Ukraine is big. Where exactly are you located?
Finally end of the day in Germany ;)

Comment (alexbarylo) [inline Services/Curl.php]: I am from Sumy, it is on north-eastern of Ukraine. Where are you located in Germany?

Comment (TobiasHauck) [inline Services/Curl.php]: Far far away :D
We are located in Koblenz, western Germany. If you want to you can add me on skype.
tobias.teeage-beatz

**Pull request closed without merging**